### PR TITLE
Upgrade to MassTransit V8

### PR DESCRIPTION
**Pull request by Dragemil**
Biggest breaking change was probably end of 3rd party DI containers support so now it relays mostly on `Microsoft.Extensions.DependencyInjection`. Tests are passing so it seems it is fine.

Be prepared for the `Newtonsoft.Json` purge, because it's high time to stop supporting both serializers.

I leave it in draft state, because it is a breaking change that needs to target v6.1 CoreLib branch which is not prepared yet.

### CODE REVIEW

**Comment by jakubfijalkowski** [comment]
image
Codecov, go home, you're drunk...

**Review by jakubfijalkowski — APPROVED** [review]

File: src/Domain/LeanCode.DomainModels.MassTransitRelay/IPipeContextServiceResolver.cs [hunk 1]
```diff
@@ -15,14 +15,14 @@ TService GetService<TService>(PipeContext pipe)
 
     public class AutofacPipeContextServiceResolver : IPipeContextServiceResolver
```
Comment (marcin-dardzinski) [inline src/Domain/LeanCode.DomainModels.MassTransitRelay/IPipeContextServiceResolver.cs]: Do we need this class anymore? It was used when we had to use service locator pattern in filters, but now we rely on MT.

File: src/Domain/LeanCode.DomainModels.MassTransitRelay/InjectLifetimeScopeFilter.cs [hunk 2]
```diff
@@ -15,19 +13,19 @@ public void Probe(ProbeContext context)
 
         public Task Send(ConsumeContext<T> context, IPipe<ConsumeContext<T>> next)
         {
-            if (!context.TryGetPayload(out Autofac.Extensions.DependencyInjection.AutofacServiceProvider scopeProvider))
+            if (!context.TryGetPayload(out Autofac.Extensions.DependencyInjection.AutofacServiceProvider? scopeProvider))
```
Comment (marcin-dardzinski) [inline src/Domain/LeanCode.DomainModels.MassTransitRelay/InjectLifetimeScopeFilter.cs]: Now I wonder if this class is necessary too. We needed it to patch MT Autofac integration, but right now it's all Microsoft.Extensions

**Review by jakubfijalkowski — CHANGES_REQUESTED** [review]
@marcin-dardzinski makes a great point - since MT does not use Autofac anymore, we should try to remove all of our Autofac or confirm that we won't be able to.

**Comment by Dragemil** [comment]
I attempted to remove everything related to MT Autofac integration and I hope it is not needed anymore.

**Comment by jakubfijalkowski** [comment]
Let's not put faith in it and add tests for that (if we don't have already) :)

**Comment by Dragemil** [comment]
Alright, bad wording - `ScopedFiltersTests` are testing that and they're passing.

**Review by jakubfijalkowski — APPROVED** [review]

**Review by marcin-dardzinski — APPROVED** [review]

**Pull request merged**